runtime/ddr/algorithm_versions.c

```diff
@@ -41,6 +41,7 @@
 #define ALG_VM_VTABLE_VERSION 1
 #define ALG_VM_ITABLE_VERSION 1
 #define ALG_VM_BYTECODE_VERSION 1
+#define OBJECT_MONITOR_VERSION 1
```

Comment:
This should probably be named `ALG_ OBJECT_MONITOR_VERSION` given the established pattern here.

Reply:
I used `OBJECT_MONITOR_VERSION` because it was already called `OBJECT_MONITOR_VERSION` in a few places like in `AlgorithmVersion.java` and `ObjectMonitor.java`. Should I change all of them to `ALG_OBJECT_MONITOR_VERSION`?

Reply:
For sanity, I'd make the constant 2 instead of 1, just so it matches the class name.

Reply:
That makes sense, but we have a long history of doing something different. We have the class `ObjectMonitor_V1` that applies to old core files. If we want to reduce confusion, I suggest we rename that class `ObjectMonitor_V0` and use `ObjectMonitor_V1` for the new one. We could apply the same renaming pattern to all the version-specific classes (in a separate piece of work).

I think we should use `ALG_OBJECT_MONITOR_VERSION` for the new version constant being introduced: the references to `OBJECT_MONITOR_VERSION` should be updated. Neither name will be in old core files so the old path will be taken.